sdk/nestjs/src/DevCycleUserGuard.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common'
+import { DevCycleUser } from '@devcycle/nodejs-server-sdk'
+
+@Injectable()
+export abstract class DevCycleUserGuard implements CanActivate {
```

Comment:
isn't this more like middleware than a guard?
https://docs.nestjs.com/middleware

it's not actually guarding anything its only function is to add stuff to the request

Reply:
Yep, although the main challenge is that would mean that any logic that's used to parse/define the user object would also need to be handled in middleware. In our case we use a guard to parse the jwt token on authenticated routes and I suspect that would often be the case.

I'd been back and forth about whether to implement this and the `RequireVariableValue` decorator as guards or interceptors, but since RequireVariableValue determines whether an endpoint can be accessed it makes the most sense as a guard.

Reply:
Would it be more of an interceptor then? It sounds like those run after guards.

I think RequireVariableValue is definitely a guard though, so I guess it needs access to the user by the time it runs..

Reply:
what about passing the user getter into the RequireVariableValue or possibly registering it beforehand in the module?